components/publish/PublishAttachment.vue

```diff
@@ -21,6 +21,74 @@ const maxDescriptionLength = 1500
 const isEditDialogOpen = ref(false)
 const description = ref(props.attachment.description ?? '')
 
+const generationInProgress = ref(false)
+
+async function generateAltText() {
+  // eslint-disable-next-line no-console
+  console.log(JSON.parse(JSON.stringify(props)))
+
+  const url = props.attachment.url
+
+  if (!url)
+    return
+
+  if (generationInProgress.value)
+    return
+
+  // TODO @Shinigami92 2024-05-27: Show confirm dialog warning message that a model with ~250MiB will be downloaded
+
+  generationInProgress.value = true
+
+  try {
+    const { pipeline, RawImage } = await import('@xenova/transformers')
+
+    const pipe = await pipeline('image-to-text', 'Xenova/vit-gpt2-image-captioning')
+
+    // const imageElement = document.querySelector<HTMLImageElement>('.dialog-main img.status-attachment-image')!
+    const imageElement = new Image()
+    imageElement.crossOrigin = 'Anonymous'
+
+    const dataUrl = new Promise<string>((resolve) => {
+      imageElement.onload = () => {
+        const canvas = document.createElement('canvas')
+        canvas.width = imageElement.width
+        canvas.height = imageElement.height
+
+        const ctx = canvas.getContext('2d')!
+        ctx.drawImage(imageElement, 0, 0)
+
+        // TODO @Shinigami92 2024-05-28: Fix "Uncaught DOMException: Failed to execute 'toDataURL' on 'HTMLCanvasElement': Tainted canvases may not be exported."
+        const dataUrl = canvas.toDataURL(`image/${url.split('.').pop()!}`)
+
+        resolve(dataUrl)
+      }
+    })
+
+    imageElement.src = `${url}?request-with-cors`
```

Comment:
move this logic before the promise and add the decode logic just after this line

Reply:
I think you don't need that logic inside the promise, the function is async